llama_index/llms/xinference.py

```diff
@@ -0,0 +1,153 @@
+            **base_kwargs,
+            **self._model_description,
+        }
+        return model_kwargs
+
+    def _get_input_dict(self, prompt: str, **kwargs: Any) -> Dict[str, Any]:
+        return {"prompt": prompt, **self._model_kwargs, **kwargs}
+
+    def chat(self, messages: Sequence[ChatMessage], **kwargs: Any) -> ChatResponse:
+        prompt = messages[-1].content if len(messages) > 0 else ""
+        history = [xinference_message_to_history(message) for message in messages[:-1]]
+        response_text = self._generator.chat(
+            prompt=prompt,
+            chat_history=history,
+            generate_config={"stream": False, "temperature": self.temperature},
+        )["choices"][0]["message"]["content"]
+        response = ChatResponse(
+            message=ChatMessage(
+                role=MessageRole.ASSISTANT,
+                content=response_text,
+            ),
+            delta=None,
+        )
+        return response
+
+    def stream_chat(
+        self, messages: Sequence[ChatMessage], **kwargs: Any
+    ) -> ChatResponseGen:
+        prompt = messages[-1].content if len(messages) > 0 else ""
+        history = [xinference_message_to_history(message) for message in messages[:-1]]
+        response_iter = self._generator.chat(
+            prompt=prompt,
+            chat_history=history,
+            generate_config={"stream": True, "temperature": self.temperature},
+        )
+
+        def gen() -> ChatResponseGen:
+            text = ""
+            for c in response_iter:
+                delta = c["choices"][0]["delta"].get("content", "")
+                text += delta
+                yield ChatResponse(
+                    message=ChatMessage(
+                        role=MessageRole.ASSISTANT,
+                        content=text,
+                    ),
+                    delta=delta,
+                )
+
+        return gen()
+
+    def complete(self, prompt: str, **kwargs: Any) -> CompletionResponse:
+        response_text = self._generator.chat(
+            prompt=prompt,
+            chat_history=None,
+            generate_config={"stream": False, "temperature": self.temperature},
+        )["choices"][0]["message"]["content"]
+        response = CompletionResponse(
+            delta=None,
+            text=response_text,
+        )
+        return response
+
+    def stream_complete(self, prompt: str, **kwargs: Any) -> CompletionResponseGen:
+        response_iter = self._generator.chat(
+            prompt=prompt,
+            chat_history=None,
+            generate_config={"stream": True, "temperature": self.temperature},
+        )
+
+        def gen() -> CompletionResponseGen:
+            text = ""
+            for c in response_iter:
+                delta = c["choices"][0]["delta"].get("content", "")
+                text += delta
+                yield CompletionResponse(
+                    delta=delta,
+                    text=text,
+                )
+
+        return gen()
```

Comment:
hmmm, I wasn't actually able to get streaming working, at least in the colab notebook.